pkg/server/commands/connected_objects.go

```diff
@@ -98,11 +98,14 @@ type ConnectedObjectsCommand struct {
 
 	// Limit limits the results yielded by the ConnectedObjects API.
 	Limit uint32
+
+	TypesystemResolver typesystem.TypesystemResolverFunc
 }
 
 func (c *ConnectedObjectsCommand) streamedConnectedObjects(
```

Comment:
Function signature is starting to get pretty bloated. Will consider some ways to make this a bit more elegant, but this works at least for the objective at hand.

Reply:
Why can't `typesys` be kept as part of `ConnectedObjectsRequest` and `reverseExpandRequest`?

Reply:
Our pattern is clear - `<something>Request` structs are for user provided inputs. The typesys is an structure constructed internally during the request lifecycle. The suggestion of embedding this in the Request input is non-standard to our codebase. You could argue that every field of the `ConnectedObjectsCommand` could/should be part of the `ConnectedObjectsRequest` as well for the same reasons you mention, but yet we don't do that.

A `ConnectedObjectsCommand` represents a specific instance of a struct that is constructed to handle a single request to an endpoint.

If we're going to do this, then we need a larger refactor for consistency. This pattern aligns with the commands/queries structs and the patterns we are using with them as they are designed today.

Reply:
> Our pattern is clear - <something>Request structs are for user provided inputs.

This is fine for the requests defined in `api`. But `ResolveCheckRequest`, `ConnectedObjectsRequest` and `reverseExpandRequest` are not defined there, we have more flexibility for these, I think.

> The suggestion of embedding this in the Request input is non-standard to our codebase.

With this specific change you're undoing the change here https://github.com/openfga/openfga/pull/650 and adding more work to yourself. Wouldn't the code work as is today?

Reply:
> This is fine for the requests defined in api. But ResolveCheckRequest, ConnectedObjectsRequest and reverseExpandRequest are not defined there, we have more flexibility for these, I think.

We are targeting a public API contract for the ConnectedObjects API. Even if we weren't, then it still is a good practice to keep things consistent. Whether an internal RPC or an external RPC it's good to keep consistent practices/semantics between public facing RPC definitions and internal facing ones. 

> With this specific change you're undoing the change here https://github.com/openfga/openfga/pull/650 and adding more work to yourself. Wouldn't the code work as is today?

#650 introduced the inconsistency. All command/query structs are constructed per request, and all other command/query structs define their parameters per struct, not per method. The change, albeit super small and hardly more work for myself, establishes more consistency within the scope of this work.

Reply:
(sidenote: if the scope of this discussion is not blocking this PR, let's resolve any other issues, merge and continue the discussion after)

Re: ConnectedObjects
Yes, we do plan to make it public. We need it in order to build a local search index without the user doing all the work.
However:
1) Not sure if `ConnectedObjects` will be the name, that was the name used internally in the API, but it may not be the best name for external usage
2) We need to discuss the interface
As in whatever gets exposed may or may not be this specific method, it will probably be a wrapper on top of it as we debate the API interface. So I wouldn't pre-optimize for that, although it is good to keep in mind that this will end up being exposed in some version or another.
For now, please do not add it to the proto files unless you hide it similar to how we did the UpdateStore, otherwise this will end up being reflected in all SDKs and docs as soon as it is merged.

My thinking is that this should be similar to all the internal methods that CheckRequest and similar wrap, it probably should not have the same inputs (middleware, etc..) that the API entry points do.
Ideally, the external interface would be consistent with the interfaces for the external methods, and the internal one would be consistent with the internal interfaces.